app/controllers/sections/firstTransporter/FirstTransporterReviewController.scala

```diff
@@ -0,0 +1,67 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.sections.firstTransporter
+
+import controllers.BaseNavigationController
+import controllers.actions._
+import forms.sections.firstTransporter.FirstTransporterReviewFormProvider
+import models.NormalMode
+import models.requests.DataRequest
+import navigation.FirstTransporterNavigator
+import pages.sections.firstTransporter.FirstTransporterReviewPage
+import play.api.data.Form
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.UserAnswersService
+import viewmodels.checkAnswers.sections.firstTransporter.FirstTransporterCheckAnswersHelper
+import views.html.sections.firstTransporter.FirstTransporterReviewView
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+class FirstTransporterReviewController @Inject()(
+                                                  override val messagesApi: MessagesApi,
+                                                  override val userAnswersService: UserAnswersService,
+                                                  override val userAllowList: UserAllowListAction,
+                                                  override val navigator: FirstTransporterNavigator,
+                                                  override val auth: AuthAction,
+                                                  override val getData: DataRetrievalAction,
+                                                  override val requireData: DataRequiredAction,
+                                                  override val withMovement: MovementAction,
+                                                  formProvider: FirstTransporterReviewFormProvider,
+                                                  val controllerComponents: MessagesControllerComponents,
+                                                  view: FirstTransporterReviewView,
+                                                  helper: FirstTransporterCheckAnswersHelper
+                                                ) extends BaseNavigationController with AuthActionHelper {
+
+
+  def onPageLoad(ern: String, arc: String): Action[AnyContent] =
+    authorisedDataRequestWithUpToDateMovementAsync(ern, arc) { implicit request =>
+      renderView(Ok, fillForm(FirstTransporterReviewPage, formProvider()))
+    }
+
+  def onSubmit(ern: String, arc: String): Action[AnyContent] =
+    authorisedDataRequestWithUpToDateMovementAsync(ern, arc) { implicit request =>
+      formProvider().bindFromRequest().fold(
+        renderView(BadRequest, _),
+        saveAndRedirect(FirstTransporterReviewPage, _, NormalMode)
```

Comment:
The underscores here are quite neat, but not knowing the code that well they're a bit confusing. Is the first underscore a different type to the second? 1st) Form[ReviewAnswer]  2nd) ReviewAnswer ?

Reply:
Exactly, the left parameter of the fold is the form (with errors). The second (right) parameter of the fold is the bound value from the form (i.e. what the user answered on the page)

Reply:
I think the code would be clearer with these values named, but happy for you to leave as is